gufe/components/proteincomponent.py

```diff
@@ -1,86 +1,224 @@
-            raise ImportError("OpenMM is currently required")
-        f = openmm.app.PDBFile(pdbfile)
-
-        return cls(f.topology, f.positions, name)
+        This is the primary serialization mechanism for this class.
 
+        """
+        Chem.MolToPDBFile(self._rdkit, pdb_file_path)
+        return pdb_file_path
+    
     @classmethod
-    def from_pdbxfile(cls, pdbxfile: str, name=""):
-        raise NotImplementedError()
-
+    def from_pdb_string(cls, pdbfile: str, name=""):
+        pass
+        
+    
     @classmethod
-    def from_rdkit(cls, rdkit: RDKitMol, name=""):
-        raise NotImplementedError()
-
-    def to_rdkit(self) -> RDKitMol:
-        raise NotImplementedError()
-
-    @classmethod
-    def from_openff(cls, offmol, name=""):
-        raise NotImplementedError()
+    def from_pdbfile(cls, pdbfile: str, name=""):
+        """
+        _summary_
 
-    def to_openff(self):
-        raise NotImplementedError()
+        Parameters
+        ----------
+        pdbfile : str
+            _description_
+        name : str, optional
+            _description_, by default ""
 
+        Returns
+        -------
+        _type_
+            _description_
+        """
+        openmm_PDBFile = PDBFile(pdbfile)
+        return cls._from_openmmPDBFile(openmm_PDBFile=openmm_PDBFile, name=name)
+    
+    
     @classmethod
-    def from_openeye(cls, oemol: OEMol, name=""):
-        raise NotImplementedError()
-
-    def to_openeye(self) -> OEMol:
-        raise NotImplementedError()
-
-    def _to_dict(self) -> dict:
-        return {'name': self.name}
+    def _from_openmmPDBFile(cls, openmm_PDBFile:PDBFile, name:str):
+        """
+        This Function serializes openmmPDBFile to 
+        AA - Protonations
+        
+        Test:        
+         - 1.5 serialization test
+         - check out files
+         - check obj
 
+        Parameters
+        ----------
+        openmm_PDBFile : PDBFile
+            _description_
+        name : str
+            _description_
+
+        Returns
+        -------
+        _type_
+            _description_
+        """              
+        periodicTable = Chem.GetPeriodicTable()
+        mol_topology = openmm_PDBFile.topology
+
+        rd_mol = Mol()
+        editable_rdmol = EditableMol(rd_mol)
+
+        # Build Topology
+        histidine_resi_atoms = {}
+        # Add Atoms
+        for atom in mol_topology.atoms():
+            atomID_orig = int(atom.index)
+            a = Atom(atom.element.atomic_number)
+            a.SetAtomMapNum(atomID_orig)
+
+            a.SetProp("name", atom.name)
+            a.SetIntProp("id", atomID_orig)
+
+            a.SetProp("resName", atom.residue.name)
+            a.SetIntProp("resId", int(atom.residue.index))
+            
+            # add residue information
+            mi  =  Chem.AtomPDBResidueInfo()
+            mi.SetResidueName(atom.residue.name)
+            mi.SetResidueNumber( int(atom.residue.index))
```

Comment:
insertion codes?

Reply:
? this is for pdb resdiue naming in the write out :)

Reply:
I moved it to the writeout - I think that makes it safer.

Reply:
ah wait I think I might not know what an insertion code is.

Reply:
added it, still don't know what it exactly does ;)